enhancements/api-review/add-new-CRD-ImageSourceDigestPolicy-and-ImageSourceTagPolicy-to-config.openshift.io.md

```diff
@@ -0,0 +1,400 @@
+
+Upgrade skew will not impact this feature. The MCO does not require skew check. CRI-O with n-2 OpenShift skew will still be able to handle the new property.
+
+### Operational Aspects of API Extensions
+
+Impacts of API extensions:
+- Components use should upgrade to use ImageSourceDigestPolicy.
+  - [Installer](https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/pkg/asset/manifests/imagesourcepolicy.go#L18)
+  - [Hypershift](https://github.com/openshift/hypershift/blob/baceec23098d39af089b06c503425c3bbee554d3/control-plane-operator/controllers/hostedcontrolplane/manifests/imagecontentsource.go)
+  - [MCO](https://github.com/openshift/machine-config-operator) The machine-config-operator/pkg/controller/container-runtime-config controller needs to operate the ImageSourceDigestPolicy and ImageSourceDigestPolicy CRDs.
+  - [Image-registry](https://github.com/openshift/image-registry/blob/a87e6c50cd973723de8b5471453de7c345403d56/pkg/dockerregistry/server/simplelookupicsp.go#L60)
+  - [Cluster-config-operator](https://github.com/openshift/cluster-config-operator)
+  - [Openshift-api-server](https://github.com/openshift/openshift-apiserver/blob/98786f917ffc7d3dc3b05893f405970b87a419b9/pkg/image/apiserver/registries/registries.go)
+  - [Runtime utils](https://github.com/openshift/runtime-utils/blob/8b8348d80d1d1e7b6cf06fb009d5965e0b55baa2/pkg/registries/registries.go#L50)
+  - [Openshift-controller-manager](https://github.com/openshift/openshift-controller-manager/blob/2a11f145ad7fcf3e92460800de1d13ba7fbb90b0/pkg/build/controller/build/build_controller.go#L20943)
+
+The impacts should be measured and done by Node team. Planning target CY21Q4
+Jira card  [OCPNODE-717](https://issues.redhat.com/browse/OCPNODE-717) will record the upgrade of ImageSourceDigestPolicy.
+
+#### Failure Modes
+
+Failure of applying new ImageSourceDigestPolicy and ImageSourceTagPolicy should not impact the cluster health.
+If there are existing ImageSourceDigestPolicy, ImageSourceTagPolicy or ImageContenSourcePolicy objects, they
+should continue to work.
+OCP Node team is likely to be called upon in case of escalation with one of the failure modes.
+
+#### Support Procedures
+
+If the cluster fails to apply new CRs, the condition of the object will be `Failed`.
+The newly added CRD will not be rolled out to the cluster.
+The machine-config-controller logs will show errors for debugging issues.
+For example, the failure of json format, the corresponding error in the log is "could not encode registries Ignition",
+
+Failure can caused by getting unexpected image when using tag pull specification when the mirror registry does not synchronize the images. Crio info level logs will show which mirror is used to pull image to help debug the problem.
+
+## Implementation History
+
+- [openshift/api#874: Create ImageContentPolicy CRD to config/v1 and allowMirrorByTags support](https://github.com/openshift/api/pull/874) PR was merged.
+## Drawbacks
```

Comment:
any drawbacks or alternatives? if no add a n/a

Reply:
I linked it to Risks and Mitigations section.